sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,369 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sunday 4th March 2018
+
+@author: Nachiket Nadkarni
+"""
+
+
+import pandas as pd
+import nibabel as nib
+import numpy as np
+from scipy.optimize import least_squares as ls
+from multiprocessing import Pool
+from functools import partial
+import tqdm
+
+
+def perf_fair_read_ptbl(perf_fair_fname):
+    """
+    Extract important information about the perfusion FAIR acquisition
+    parameters from the *_ptbl.txt file created by
+    sammba.io_conversions.dcm_to_nii.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the perfusion *.nii.gz file converted by
+        sammba.io_conversions.dcm_to_nii, NOT to the *_ptbl.txt file itself.
+        The *_ptbl.txt file name is determined automatically fro; that of the
+        *.nii.gz.
+
+    Returns
+    -------
+    Dictionary of parameters useful to perfusion fitting functions.    
+    """
+    
+    # the *_ptbl.txt file will have the same name as the perfusion file itself,
+    # but with _ptbl.txt at the end rather than just .nii.gz. arguably this
+    # function should directly take the text file name as input, but it is
+    # difficult to envisage a scenario where this would be needed. the
+    # *_ptbl.txt file has a specific format created by
+    # sammba.io_conversions.dcm_to_nii. if someone supplies their own file
+    # name, it would mean they changed the *_ptbl.txt name (why?) or they have
+    # their own formatted text file which this function would not be able to
+    # process
+    
+    bfx = perf_fair_fname.split('_')[-1].split('.')[0]
+    ptbl = pd.read_table(perf_fair_fname.replace(bfx + '.nii.gz', bfx + '_ptbl.txt'))
+    ti = ptbl.TI[ptbl.slice == 1][ptbl.FC == 'Selective Inversion'].tolist()

```

Comment:
may be at some point a check function could be coded, that makes sure fields `TI`, `slice` etc exist and throws a meaningful error instead. This will avoid weird errors if parsing an uncorrect file